### PR TITLE
Removed installer creation and cleanup

### PR DESCRIPTION
**Pull request by soumeh01**
- Workflow cleanup (Removed cmiss-toolbox.sh dependencies)
- Repo cleanup
- Removed gtest-based tests
- Added new tests

Addresssing https://github.com/Open-CMSIS-Pack/devtools/issues/1071

### CODE REVIEW

File: .github/workflows/toolbox.yml [hunk 1]
```diff
@@ -92,224 +91,339 @@ jobs:
...
+          cp distribution/bin/cpackget.mac-arm64  zip/cmsis-toolbox-darwin-arm64/bin/cpackget
+          cp distribution/bin/csolution.mac-arm64 zip/cmsis-toolbox-darwin-arm64/bin/csolution
+          cp distribution/bin/packchk.mac-arm64   zip/cmsis-toolbox-darwin-arm64/bin/packchk
+          cp distribution/bin/svdconv.mac-arm64   zip/cmsis-toolbox-darwin-arm64/bin/svdconv
+          cp distribution/LICENSE.txt             zip/cmsis-toolbox-windows-amd64/LICENSE.txt
+          cp distribution/LICENSE.txt             zip/cmsis-toolbox-windows-arm64/LICENSE.txt
+          cp distribution/LICENSE.txt             zip/cmsis-toolbox-linux-amd64/LICENSE.txt
+          cp distribution/LICENSE.txt             zip/cmsis-toolbox-linux-arm64/LICENSE.txt
+          cp distribution/LICENSE.txt             zip/cmsis-toolbox-darwin-amd64/LICENSE.txt
+          cp distribution/LICENSE.txt             zip/cmsis-toolbox-darwin-arm64/LICENSE.txt
```
Comment (brondani) [inline .github/workflows/toolbox.yml:143]: The `etc/setup` file is only for standalone Build Manager usage compatibility and should not be part of the toolbox archives.         
```suggestion
          cp distribution/LICENSE.txt             zip/cmsis-toolbox-darwin-arm64/LICENSE.txt
          rm distribution/etc/setup
```

File: test/tests.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    dirDict["doc"] = {'index.html'}
+    dirDict["etc"] = {
+        "AC6.6.18.0.cmake", "ac6_linker_script.sct", "gcc_linker_script.ld",
+        "iar_linker_script.icf", "CPRJ.xsd", "PACK.xsd", "GCC.10.3.1.cmake",
+        "CLANG.16.0.0.cmake", "IAR.9.32.1.cmake",
+        "{{ProjectName}}.cproject.yml", "{{SolutionName}}.csolution.yml",
+        "clayer.schema.json", "common.schema.json", "cproject.schema.json",
+        "cproject.schema.json", "CMakeASM",
+        "cbuild.schema.json", "cdefault.schema.json", "cdefault.yml",
+        "clang_linker_script.ld", "CMSIS-Build-Utils.cmake", "csolution.schema.json"}
```
[the remark this answers is not in the document]
Comment (soumeh01) [reply test/tests.py]: fixed [Alphabatical sorting of files](https://github.com/Open-CMSIS-Pack/cmsis-toolbox/pull/60/commits/0997bd36ac3ab79c55f2c0eb409debd71a76832f)

File: .github/workflows/toolbox.yml (hunk 1 above)
Comment (soumeh01) [reply .github/workflows/toolbox.yml:137]: fixed [remove setup file](https://github.com/Open-CMSIS-Pack/cmsis-toolbox/pull/60/commits/e30513329d3da49f57b9858184735f4d7f044125)

**Review by jkrech — APPROVED** [review]
Do we need to test file permissions to have executable rights for binaries?

**Comment by jkrech** [comment]
Do we have a related issue for this PR that we should add to the scope of Version 2.1.0?

**Comment by soumeh01** [comment]
> Do we have a related issue for this PR that we should add to the scope of Version 2.1.0?

I would say this change is the leftover path of https://github.com/Open-CMSIS-Pack/devtools/issues/957

**Comment by jkrech** [comment]
Please create a new one, since the other got closed for 2.0.0.

**Comment by soumeh01** [comment]
> Do we have a related issue for this PR that we should add to the scope of Version 2.1.0?



> Do we need to test file permissions to have executable rights for binaries?

Good point, let me check and investigate on that.

File: .github/workflows/toolbox.yml [hunk 3]
```diff
@@ -85,12 +85,6 @@ jobs:
           cp ../docs/index.html distribution/doc
         working-directory: toolbox
 
-      - name: Add execution permission to the binaries and set files ownership
-        run: |
-          sudo chmod -R +x bin/*
-          sudo chown -R root:root *
-        working-directory: toolbox/distribution
-
```
[the remark this answers is not in the document]
Comment (brondani) [reply .github/workflows/toolbox.yml]: https://github.com/Open-CMSIS-Pack/cmsis-toolbox/pull/61

**Pull request merged**